.github/workflows/lint.yml

```diff
@@ -0,0 +1,62 @@
+name: Reusable Linting
+
+on:
+  workflow_call:
+    inputs:
+      node-version:
+        required: false
+        default: "20"
+        type: string
+      php-version:
+        required: false
+        default: "7.4"
+        type: string
+      validate-php-phpmd:
+        required: false
+        default: false
+        type: boolean
+    secrets:
+      token:
+        required: true
+
+jobs:
+  execute:
+    runs-on: ubuntu-22.04
+    name: Execute
+    timeout-minutes: 30
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - id: files
+      uses: studyportals/get-changed-files@main
+      with:
+        token: ${{ secrets.token }}
+        format: csv
+        extensions: ".php"
+      if: inputs.validate-php-phpmd
+    - uses: actions/setup-node@v4
+      with:
+        node-version: ${{ inputs.node-version }}
+    - uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ inputs.php-version }}
+      if: inputs.validate-php-phpmd
+    - run: npm ci @studyportals/code-style@$(jq -r '.devDependencies."@studyportals/code-style"' package.json) --no-save
+    - run: |
+        composer config github-oauth.github.com ${{ secrets.token }}
+        npm run configure || true
```

Comment:
Is the `true` needed here?

Reply:
I added it specifically for PHP projects that don't have the `configure` command. As far as I can tell, only Portal has it. If you think it is not necessary based on my reasoning, I can drop it.

Reply:
Why is `npm run configure` is necessary even for Portal?

Reply:
In order for `composer install` or `composer update` to work, you have to run the `configure` command. I believe it is related to the `cghooks` that are run after the `install/update` command.

Reply:
configure.sh is a script in portal which copies shared code base

Reply:
I know what it does. I'm wondering why is that needed for linting

Reply:
It is needed because the PHPMD package needs to be installed. To do so without encountering an error, `configure` has to be run based on my testing.

Reply:
phpmd is probably installed through composer install. I'm wondering why running bash file that copies the code codebase associated to flavours and environments would block you from install phpmd. If that happens, perhaps the issue lies elsewhere

Reply:
It blocks because after the install happens, post-install hooks are run. Those hooks expect those files, hence the install fails with an error code. @skrothapalli1, do you know if you can skip the post-install hooks? I briefly tried to see if you could, but didn't find anything.

Reply:
`composer install --no-scripts` might be an option

Reply:
I tried that, but it does not work. I get the same error as before. Here is the [error](https://github.com/studyportals/Portal/actions/runs/9265484772/job/25487616554?pr=5179) I get

Reply:
ok

Reply:
environment.php and config.php are copied by the configure.sh. These files are needed by php as they are being required by Index.php. 
If we can copy these files into the codebase (Config folder) then we would not need to run `npm run configure` command

Reply:
Do you mean I should put empty `environment.php` and `config.php` files in that directory?

Reply:
If we copy those files then we do not need to run the npm run configure in the linter.yml but it is still need to be run at other places during deployment

Reply:
not empty files, actual files from either development or testing. To avoid confusion, lets not do this for now

Reply:
Alright